tests/rptest/tests/archival_test.py

```diff
@@ -376,6 +385,40 @@ def check_upload():
 
         validate(check_upload, self.logger, 90)
 
+    @cluster(num_nodes=3)
+    def test_retention_archival_coordination(self):
+        """
+        Test that only archived segments can be evicted and that eviction
+        restarts once the segments have been archived.
+        """
+        self.kafka_tools.alter_topic_config(
+            self.topic,
+            {
+                TopicSpec.PROPERTY_RETENTION_BYTES: 5 * self.log_segment_size,
+            },
+        )
+
+        with firewall_blocked(self.redpanda.nodes, self._get_s3_endpoint_ip()):
+            produce_until_segments(redpanda=self.redpanda,
+                                   topic=self.topic,
+                                   partition_idx=0,
+                                   count=10)
+
+            # Sleep some time sufficient for log eviction under normal conditions
+            # and check that no segment has been evicted (because we can't upload
+            # segments to the cloud storage).
+            time.sleep(3 * self.log_compaction_interval_ms / 1000.0)
+            counts = segments_count(self.redpanda, self.topic, partition_idx=0)
+            self.logger.info(f"node segment counts: {list(counts)}")
+            assert all(c == 10 for c in counts)
```

Comment:
`produce_until_segments` may produce more than `count` segments, so we should check for `c >= 10`.